apps/snitch_core/lib/core/domain/shipment_engine.ex

```diff
@@ -1,66 +1,77 @@
 defmodule Snitch.Domain.ShipmentEngine do
   @moduledoc """
-  Implements functions to handle creation of the optimal shipment.
+  Finds the optimal shipment for a given order.
 
-  > Note 
-    No support for partially fulfilled and backordered packages.
+  The current module returns `shipments` under the following constraints:
+  * Selected `shipments` won't fulfill the same `lineitem`.
```

Comment:
We provide only _one `shipment`_. The `shipment` has many `packages`. Hence:
1. Two (or more) packages will not fulfill the same line-item
2. The shipment as a whole will fulfill the complete order.

It's more important to highlight that we don't give a damn about backorder. Please refrain from "current" module, and mention that the shipment is found using a CSP.